sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -382,3 +382,27 @@ function buildRawAuthorizationRule(authorizationRule: AuthorizationRule): any {
   };
   return rawAuthorizationRule;
 }
+
+/**
+ * @ignore
+ * Recurse through a JSON structure and remove key: value pairs where the value == `undefined`.
+ * Note: This utility must be used with caution as it mutates the passed in object.
+ * @param value The JSON structure
+ * @returns The new JSON structure with undefined values removed
+ */
+export function removeUndefinedValues(value: any): any {
+  if (Array.isArray(value)) {
+    value.forEach((entry) => {
+      removeUndefinedValues(entry);
+    });
+  } else if (Object.prototype.toString.call(value) === "[object Object]") {
+    Object.keys(value).forEach(function(property) {
+      if (value[property] == undefined) {
+        delete value[property];
+      } else if (Object.prototype.toString.call(value[property]) === "[object Object]") {
```

Comment:
Does this also need to recursively remove undefined/null from any array properties? It looks like you're recursively removing undefined/null values from objects, but only remove undefined/null values from arrays if the original value is an array due to this check.

Reply:
Yes actually, I was thinking about it too based on some comments on xml2js issue.
Updated to remove these since they would end up in generating empty tags as well.

Reply:
Correction.. only the top-level elements are not allowed to be empty. I had a suspicion on this initially but is confirmed based on test runs - the inner properties can have empty tags and the service processes them just fine. This might as well be a bug filed on the service as they should accept valid xml content, or atleast handle stuff uniformly.

Updated helper and documentation accordingly.

cc: @bterlson @ramya-rao-a @AlexGhiondea

Reply:
Also, since the input `resource` is never an array, and it isn't certain how the service would treat empty tags in it should the feature be extended in future, removed the utility and inlined the changes to remove just the top level key: value pairs.